src/main/java/com/powsybl/openloadflow/util/Reports.java

```diff
@@ -392,12 +388,33 @@ public static void reportNewtonRaphsonLargestMismatches(Reporter reporter, Strin
 
         ReportBuilder busInjectionReportBuilder = Report.builder();
         busInjectionReportBuilder.withKey("NRMismatchBusInjection")
-                .withDefaultMessage("Bus injection: ${busSumP} MW, ${busSumQ} MVar")
-                .withValue("busSumP", busReport.p())
-                .withValue("busSumQ", busReport.q())
+                .withDefaultMessage("Bus injection: ${busP} MW, ${busQ} MVar")
+                .withValue("busP", busReport.injectionP())
+                .withValue("busQ", busReport.injectionQ())
                 .withSeverity(TypedValue.TRACE_SEVERITY);
 
-        Reporter subReporter = reporter.createSubReporter("NRMismatch", "Largest ${equationType} mismatch: ${mismatch}", subReporterMap);
+        String mismatchUnit;
+        double mismatchUnitConverter;
+        switch (acEquationType) {
+            case "P" -> {
+                mismatchUnit = "MW";
+                mismatchUnitConverter = PerUnit.SB;
+            }
+            case "Q" -> {
+                mismatchUnit = "MVar";
+                mismatchUnitConverter = PerUnit.SB;
+            }
+            default -> {
+                mismatchUnit = "p.u.";
+                mismatchUnitConverter = 1.0;
+            }
+        }
+
+        Map<String, TypedValue> subReporterMap = new HashMap<>();
+        subReporterMap.put("equationType", new TypedValue(acEquationType, TypedValue.UNTYPED));
+        subReporterMap.put("mismatch", new TypedValue(mismatchUnitConverter * busReport.mismatch(), OpenLoadFlowReportConstants.MISMATCH_TYPED_VALUE));
+        subReporterMap.put("mismatchUnit", new TypedValue(mismatchUnit, TypedValue.UNTYPED));
+        Reporter subReporter = reporter.createSubReporter("NRMismatch" + acEquationType, "Largest ${equationType} mismatch: ${mismatch} ${mismatchUnit}", subReporterMap);
```

Comment:
I cannot understand the reason of the change of key from just `"NRMismatch"` to now `"NRMismatch" + acEquationType` ? Why do you need a different key, while the formatting message is always the same ?

Reply:
You're right. I guess I thought that since the units would be different, so should the reporter name. Just fixed it.